uint32 current
string name